assets/js/modules/tagmanager/hooks/useGAPropertyIDEffect.js

```diff
@@ -32,11 +32,21 @@ export default function useGAPropertyIDEffect() {
 	const singleAnalyticsPropertyID = useSelect( ( select ) =>
 		select( MODULES_TAGMANAGER ).getSingleAnalyticsPropertyID()
 	);
+	const tagmanagerExistingSettings = useSelect( ( select ) =>
+		select( MODULES_TAGMANAGER ).getSettings()
+	);
 	const { setGAPropertyID } = useDispatch( MODULES_TAGMANAGER );
 
 	useEffect( () => {
-		if ( singleAnalyticsPropertyID !== undefined ) {
+		if (
+			!! tagmanagerExistingSettings &&
```

Comment:
This is fine as once set, settings will be an object, but is there any reason not to use the slightly more explicit and clear `!== undefined` here? It just might make things a bit more readable, as that's the only non-truthy value we'd expect here, and our convention in code is that `undefined` means "loading"/"not yet loaded", so it makes it a bit more clear just from the condition that `tagmanagerExistingSettings` needs to be loaded, but otherwise the value doesn't matter.

```suggestion
			tagmanagerExistingSettings !== undefined &&
```